SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -105,11 +105,44 @@ void USpatialReceiver::LeaveCriticalSection()
 		{
 			OnEntityAddedDelegate.Broadcast(PendingAddEntity);
 		}
+		PendingAddComponents.RemoveAll([PendingAddEntity](const PendingAddComponentWrapper& Component) {return Component.EntityId == PendingAddEntity;});
 	}
 
 	for (Worker_AuthorityChangeOp& PendingAuthorityChange : PendingAuthorityChanges)
 	{
-		HandleActorAuthority(PendingAuthorityChange);
+		if (PendingAuthorityChange.authority != WORKER_AUTHORITY_AUTHORITATIVE)
+		{
+			HandleActorAuthority(PendingAuthorityChange);
+		}
+	}
+
+	for (PendingAddComponentWrapper& PendingAddComponent : PendingAddComponents)
+	{
+		if (StaticComponentView->GetComponentData<UnrealMetadata>(PendingAddComponent.EntityId) == nullptr)
```

Comment:
Should we log in these cases? Does this mean we don't have the actor checked out?

Reply:
I have started writing a really long explanation, and I am no longer sure whether this is correct.
The initial reasoning for this was that I wanted to avoid handling components we didn't need to handle. `HandleIndividualAddComponent` is written in a way that assumes we are dealing with dynamic components on actors. The code above therefore discards all add components which are not associated with actors (e.g. GlobalStateManager, SpatialSpawner, etc.).
This should mostly work, but also, it should not happen.
If `HandleIndividualAddComponent` finds out that the component doesn't have an associated actor channel, it prints a warning. I think I will just remove this condition and let it fall into the logging branch of `HandleIndividualAddComponent`. But this does make me uneasy.